File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.*;


class Solution {
    public int[] solution(int[] arr, int divisor) {
        List<Integer> list = new ArrayList<Integer>();
        Arrays.sort(arr);
        
        for(int num : arr){
            if(num % divisor == 0 ){
                list.add(num);
            }
        }
        
        int listSize = list.size();
        
        if(listSize == 0){
            int[] answer = {-1};
            return answer;
        }
        
        int[] answer = new int[listSize];
        for(int i = 0 ; i < listSize ; i++){
            answer[i]=list.get(i);            
        }
        return answer;
    }
}

File: C#/프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.cs
using System.Linq;
public class Solution {
    public int[] solution(int[] arr, int divisor) {
            int[] answer = arr.Where(x => x % divisor == 0)
                                .OrderBy(x => x)
                                .DefaultIfEmpty(-1)
                                .ToArray();



            return answer;
    }
}